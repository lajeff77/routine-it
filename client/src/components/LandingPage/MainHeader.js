import { AppBar, Button, Toolbar, Typography } from '@mui/material';
import { Link } from 'react-router-dom';
import React from 'react';

const MainHeader = () => {

    return(
        <AppBar position="static" sx={{padding: "1%"}}>
            <Toolbar>
                <Typography variant='h3' component="div" sx={{ flexGrow: 1 }}>Routine It</Typography>            
                <Button component={Link} to="/" variant="contained" disableElevation sx={{margin: "10px"}}>Home</Button>
                <Button component={Link} to="/signup" variant="contained" disableElevation sx={{margin: "10px"}}>Sign Up</Button>
                <Button component={Link} to="/signin" variant="contained" disableElevation sx={{margin: "10px"}}>Sign In</Button>
            </Toolbar>
        </AppBar>
    )
}

export default MainHeader;