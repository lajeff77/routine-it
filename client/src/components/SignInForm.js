import React from "react";
import {Link} from 'react-router-dom';

const SignInForm = () => {

    const handleSubmit = (e) => {
        e.preventDefault();
        /*TODO: call account login api here */
        console.log("Signed in account");
    }

    return(
        <div>
        <body className='form-body'>
         <form className='login-signin-form' onSubmit={handleSubmit}>
                <h2>Welcome Back to Routine It</h2>
                <p>Need to create an account? Sign up <Link to="/signup" className="regular-link">here</Link>.</p>
                <div id="create-account-form-inputs">
                    <label>Email</label>
                    <input type="email" name="email"/>
                    <label>Password</label>
                    <input type="password" name="password"/>
                </div>
                <input type="submit" value="Sign In" className="button"/>
            </form>
        </body>
    </div>
    )

}

export default SignInForm;