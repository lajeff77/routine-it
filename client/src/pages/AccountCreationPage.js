import React from 'react';
import MainHeader from '../components/MainHeader';
import Footer from '../components/Footer';

const AccountCreationPage = () => {

    return (
        <div>
            <MainHeader />
            <Footer />
        </div>
    )
}

export default AccountCreationPage;