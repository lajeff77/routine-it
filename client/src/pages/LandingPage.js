import React from 'react';
import MainHeader from '../components/LandingPage/MainHeader';
import HeroSection from '../components/LandingPage/HeroSection';
import SummarySection from '../components/LandingPage/SummarySection';
import Footer from '../components/LandingPage/Footer';

const LandingPage = () => {

    return (
        <div>
            <MainHeader/>
            <HeroSection/>
            <SummarySection/>
            <Footer />
        </div>
    )
}

export default LandingPage;