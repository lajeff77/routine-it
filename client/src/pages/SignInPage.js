import React from 'react';
import MainHeader from '../components/MainHeader';
import Footer from '../components/Footer';

const SignInPage = () => {

    return (
        <div>
            <MainHeader />
            <Footer />
        </div>
    )
}

export default SignInPage;