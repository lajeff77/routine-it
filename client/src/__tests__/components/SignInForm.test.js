import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import LandingPage from '../../pages/LandingPage';
import AccountCreationPage from '../../pages/AccountCreationPage';
import SignInPage from "../../pages/SignInPage";
import SignInForm from "../../components/SignInForm";

describe(SignInForm.name, ()=>{
    test('SignInForm matches snapshot', () => {
        const { asFragment } = render(
          <MemoryRouter>
            <Routes>
              <Route element={<LandingPage />} path="/" />
              <Route element={<AccountCreationPage />} path="/signup" />
              <Route element={<SignInPage />} path="/signin" />
            </Routes>
          </MemoryRouter>
        );
    
        expect(asFragment()).toMatchSnapshot();});
});