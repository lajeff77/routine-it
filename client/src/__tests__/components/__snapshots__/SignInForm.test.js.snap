// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm SignInForm matches snapshot 1`] = `<DocumentFragment />`;
